src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int START_ARRAY_CAPACITY = 10;
+    private static final double CAPACITY_ARRAY_MULTIPLE = 1.5;
+    private int size;
+    private T[] stackArray;
+    private int capacity;
 
+    public MateStack() {
+        capacity = START_ARRAY_CAPACITY;
+        stackArray = (T[]) new Object[capacity];
     }
 
-    public T peek() {
-        return null;
+    public void push(T value) {
+        checkArraySize();
+        stackArray[size] = value;
+        size++;
     }
 
-    public T pop() {
-        return null;
+    public T peek() throws EmptyStackException {
+        if (size > 0) {
+            return stackArray[size - 1];
+        }
+        throw new EmptyStackException();
```

Comment:
create separate method for checking size

Reply:
done